src/net/smooth/index.ts

```diff
@@ -201,8 +203,9 @@ export default function(
       }
 
       const { mimeType } = representation;
-      const base = resolveURL(representation.baseURL);
-      const url = buildSegmentURL(base, representation, segment);
+      const baseURL = resolveURL(segment.media);
+      const intermediateURL = replaceRepresentationSmoothTokens(baseURL, representation);
+      const url = replaceSegmentSmoothTokens(intermediateURL, segment.time);
```

Comment:
Why not a single ``replaceSmoothTokens`` function?

Reply:
Actually, why does ``media`` contains tokens here?

I thought the purpose was to have the complete URL directly. Here it looks more like some private data the transport needs to construct its URL.